packages/types/types/ts3.6/index.d.ts

```diff
@@ -2,7 +2,9 @@ export interface Action<T extends string = string> {
   type: T
 }
 
-export type Saga<Args extends any[] = any[]> = (...args: Args) => Iterator<any>
+export type Saga<Args extends any[] = any[], TReturn = any> = (...args: Args) => Iterator<any, TReturn>
+
+export type SagaReturnType<S extends Saga<any, any>> = S extends Saga<any, infer R> ? R : never
```

Comment:
There is a case in which we might get `never` back from this. I don't see any particular problems with this... but it's fine somewhat coincidentally. It's fine because the computed return type is only used within `toPromise` and `result`.

It's fine for `toPromise` as a never-returning iterator should, well, never complete - so the returned promise won't ever resolve and thus it doesn't quite matter.

It's fine for `result` because `T` is there included within a union with `undefined`, `never` is always eliminated from unions, so the only leftover will always be just `undefined` in such a case and that's correct.

This [TS playground](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mAVAAhmYBTATljATAYTnCmwAoBKZAbwChkHkAbDKVZAXmQAZ7GB3ABYwWyMlCwgMVOoznIAnjAxM8qANTrkfBgF9a+2hAQBnNjDZc0mHPiIlyFWrVCRYCFAEMwYOFE+khMRgpFiUAFzIAOIYYNgBcFgAPGAgALYARtgANMhxAG7YAHw0Osys7Fy8ckIiGGISUjJlckoqajCa2nL6hsZgZqhQAEycyN6+-oH2IY7OUAoADvUAyp4A5p5JAIJY6ybIGAAepGB4B94KANoAumOXt7kAKgBKrCBYYPdgCiVcZAA6IGePYmSK7fZUDglACSoQSyUuzzeUA+YCK8yWqw2nhRaKeWKSK0OJ1i52Qa02SSR4x+RT+FJJp3JlK2NOs2GQLxKAH4uchIgVsJjlsgnhgzABGMasvGfAnLJILZZwYCodDYXBBBxYEoAej1yH4iQA1iZaAbGAA9Hki+risyjLiy97ywnKjCq2mTAJ2YKhfWGvBwCVDZAQXC+2kKZCLLBwDIsNIHYCJZAgEy2y0MG3OWh4DAQJgg+quaDwL6eCAQDCLKArCRodZkMxYSKtpsUSL5OAwPDOKs1usNrBNsjUXTjA5CrBOWiD2v1xtgZsTqd5DCFLDIAA+6bOGGAaHwFCAA) might show what kind of edge cases I was thinking about while reviewing this change.

I just wonder if we should have any tests to check against potential regressions here. 🤔 Maybe I'm overthinking this though.

It's also worth noting that `T` in `Task` is covariant and thus `Saga<any, never>` satisfies `Saga<any, any>`. It's always worth re-checking this because the same is not true for contravariant parameters: [TS playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgMIHsBucrDuAHnQFcxkAVAPgG8AoZB5MdABSnQFtgBnCALmQAKAJTIAvJWRtOPCASq0AvrVpgAngAcUAQXFosOPIRARM0SRAAekEABNu+7LnxgC+NZID8yAIzIBAAy0APTBjAB6niqgkLCIKADKYMAANikYTkauJGRUdIxMrOxcvAIi4pLSJXIKyqqaKABCekmp6QbOxqbmyFY29sitaRmGLm4gHsjefoEhYQyR0eDQ8Ej64FBwmWOgFDT0jAhwaWVwAuSiEsiY6MC2SiqhTAAWPMhHIMgARiga7F8pCAcODJBD1LRoPQYDZbUZdMxQCzWCB2BzQsCbbaEdxeXz+ZBBJ4LTxAA)